internal/scheduler/scheduler.go

```diff
@@ -140,17 +138,33 @@ func (s *Scheduler) UpdateJobNextRun(ctx context.Context, jobId JobId, nextRunIn
 }
 
 // Start begins the scheduling loop that will query the repository for jobs to run and
-// run them in a goroutine.
-func (s *Scheduler) Start(ctx context.Context) {
+// run them in a goroutine, the scheduler will stop all running jobs and stop requesting
+// new jobs once the ctx past in is cancelled.
+// The scheduler cannot be started again once the ctx is cancelled, a new scheduler will
+// need to be instantiated in order to begin scheduling again.
+func (s *Scheduler) Start(ctx context.Context) error {
+	const op = "scheduler.(Scheduler).Start"
 	if !s.started.CAS(s.started.Load(), true) {
 		s.logger.Debug("scheduler already started, skipping")
-		return
+		return nil
+	}
+
+	repo, err := s.jobRepoFn()
+	if err != nil {
+		return errors.Wrap(err, op)
+	}
+
+	// Interrupt all runs previously allocated to this server
+	_, err = repo.InterruptRuns(ctx, 0, job.WithServerId(s.serverId))
+	if err != nil {
+		return errors.Wrap(err, op)
 	}
 
-	s.baseContext, s.baseCancel = context.WithCancel(ctx)
+	s.baseContext = ctx
```

Comment:
This was a change from having a next context created here and it was a simple change to just do `s.baseContext = ctx` so went with it.  Now that you bring it up I could just pass this context into the goroutines that need it instead of pulling it from the struct! ty

Reply:
Just did this, so clean TY!